fvtest/compilertest/tests/CCDataTest.cpp

```diff
@@ -0,0 +1,639 @@
+#include "OptTestDriver.hpp"
+
+using namespace TR;
+using namespace TestCompiler;
+
+class TableInjector : public TR::IlInjector
+   {
+   public:
+
+   TR_ALLOC(TR_Memory::IlGenerator)
+
+   TableInjector(TR::TypeDictionary *types, TestDriver *test, const int32_t * const caseValues, const size_t numCaseValues)
+   : _caseValues(caseValues), _numCaseValues(numCaseValues), TR::IlInjector(types, test)
+      {
+      }
+
+   bool injectIL()
+      {
+      createBlocks(2 + _numCaseValues);
+
+      generateToBlock(0);
+      auto defaultCase = TR::Node::createCase(NULL, block(1)->getEntry());
+      auto table = TR::Node::create(TR::table, 2 + _numCaseValues, parameter(0, Int32), defaultCase);
+      for (auto i = 0; i < _numCaseValues; ++i)
+         table->setAndIncChild(2 + i, TR::Node::createCase(NULL, block(2 + i)->getEntry()));
+      genTreeTop(table);
+      cfg()->addEdge(_currentBlock, block(1));
+      for (auto i = 0; i < _numCaseValues; ++i)
+         cfg()->addEdge(_currentBlock, block(2 + i));
+
+      generateToBlock(1);
+      returnValue(parameter(1, Int32));
+
+      for (auto i = 0; i < _numCaseValues; ++i)
+         {
+         generateToBlock(2 + i);
+         returnValue(iconst(_caseValues[i]));
+         }
+
+      return true;
+      }
+
+   private:
+      const int32_t  *_caseValues;
+      const size_t   _numCaseValues;
+   };
+
+class GeneratedMethodInfo : public MethodInfo
+   {
+   public:
+      typedef int32_t (*compiled_method_t)(int32_t, int32_t);
+
+      GeneratedMethodInfo(TestDriver *test, const int32_t * const caseValues, const size_t numCaseValues) : _ilInjector(&_types, test, caseValues, numCaseValues)
+         {
+         TR::IlType *int32 = _types.PrimitiveType(TR::Int32);
+         _args[0] = {int32};
+         _args[1] = {int32};
+         DefineFunction(__FILE__, LINETOSTR(__LINE__), "TableTest", sizeof(_args) / sizeof(_args[0]), _args, int32);
+         DefineILInjector(&_ilInjector);
+         }
+   private:
+      TR::TypeDictionary   _types;
+      TR::IlType           *_args[2];
+      TableInjector        _ilInjector;
+   };
+
+class TableTest : public OptTestDriver
+   {
+   public:
+      TableTest() : _caseValues(NULL), _numCaseValues(0) {}
+      void setCaseValues(const int32_t * const caseValues, const size_t numCaseValues)
+         {
+         _caseValues = caseValues;
+         _numCaseValues = numCaseValues;
+         }
+      virtual void invokeTests() override
+         {
+         EXPECT_TRUE(_caseValues != NULL);
+         EXPECT_GT(_numCaseValues, 0);
+         auto compiledMethod = getCompiledMethod<GeneratedMethodInfo::compiled_method_t>();
+         for (auto i = 0; i < _numCaseValues; ++i)
+            EXPECT_EQ(compiledMethod(i, -1), _caseValues[i]);
+         EXPECT_EQ(compiledMethod(_numCaseValues, -1), -1);
+         }
+      private:
+         const int32_t  *_caseValues;
+         size_t         _numCaseValues;
+   };
+
+class TableVerifier : public TR::IlVerifier
```

Comment:
What's the purpose of this verifier?

Reply:
As of this PR, nothing. On the snapshot branch, where the x86 code generator uses CCData in its table evaluator when the right options are in effect, it generates code that will exercise CCData, even though an IL verifier can't actually test for it.

From a development POV it lets you quickly generate a compilation log and a debuggable binary that incorporate CCData so that you can at least verify things by inspection, which is better than nothing. I haven't kept up with the OMR compiler's testing framework, so if there's a way to actually verify that the generated code has certain features I'd be happy to use it.